### PR TITLE
[ASM] Single Step ASM Helm chart

### PR DESCRIPTION
**Pull request by eliottness**
### What does this PR do? What is the motivation?

Cherry-pick from #22371 

Document support about ASM Single Step Installation for Kubernetes Helm Chart

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 1]
```diff
@@ -9,7 +9,8 @@ kind: documentation
 ## Requirements
 
 - **Minimum Agent version 7.53.0**
-- **Languages and architectures**: Single step ASM instrumentation for code security only supports tracing Java, Node.js, Python (private beta), and .NET Core services on `x86_64` and `arm64` architectures.
+- **Minimum Helm version 3.62.0** (For Kubernetes deployments)
```
Comment (eliottness) [inline content/en/security/application_security/enabling/single_step/code_security.md:12]:
```suggestion
- **Minimum Datadog Helm chart version 3.62.0** (For Kubernetes deployments)
```

**Comment by brett0000FF** [comment]
Thanks! Created a card for our team to review: DOCS-8006.

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 2]
```diff
@@ -168,10 +168,72 @@ docker run -d --name dd-agent \
...
 
+
 {{% /tab %}}
 
-{{< /tabs >}}
+{{% tab "Kubernetes" %}}
+
+You can enable APM by installing the Agent with the Datadog Helm chart. This deploys the Datadog Agent across all nodes in your Linux-based Kubernetes cluster with a DaemonSet.
+
+**Note**: Single step instrumentation doesn't instrument applications in the namespace where you install the Datadog Agent. It's recommended to install the Agent in a separate namespace in your cluster where you don't run your applications.
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:178]:
```suggestion
<div class="alert alert-info">Single step instrumentation doesn't instrument applications in the namespace where you install the Datadog Agent. It's recommended to install the Agent in a separate namespace in your cluster where you don't run your applications.</div>
```

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 3]
```diff
@@ -168,10 +168,72 @@ docker run -d --name dd-agent \
...
+### Installation
+
+To enable single step instrumentation with Helm:
+
+1. Add the Helm Datadog repo:
+   ```bash
+    helm repo add datadog https://helm.datadoghq.com
+    helm repo update
+    ```
+2. Create a Kubernetes secret to store your Datadog [API key][10]:
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:193]:
```suggestion
2. Create a Kubernetes Secret to store your Datadog [API key][10]:
```

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 4]
```diff
@@ -168,10 +168,72 @@ docker run -d --name dd-agent \
...
+   ```
+   Replace `<DATADOG_SITE>` with your [Datadog site][12].
 
+4. Run the following command:
+   ```bash
+   helm install datadog-agent -f datadog-values.yaml datadog/datadog
+   ```
+5. Do a rolling restart on your applications for instrumentation to take effect.
+
+More information on Kubernetes Single Step Instrumentation here:
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:217]:
```suggestion
For more information on Kubernetes single step instrumentation, see the following:
```

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 5]
```diff
@@ -208,6 +270,26 @@ Run the following commands and restart the service to stop injecting the library
...
+     template:
+       metadata:
+         labels:
+           admission.datadoghq.com/enabled: "false"
+   ```
+2. Apply the configuration:
+   ```bash
+   kubectl apply -f /path/to/your/deployment.yaml
+   ```
+3. Restart the services you want to remove instrumentation for.
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:287]:
```suggestion
1. Set the `admission.datadoghq.com/enabled:` label to `"false"` for the pod spec:
  
   ```yaml
   spec:
     template:
       metadata:
         labels:
           admission.datadoghq.com/enabled: "false"
   ```
2. Apply the configuration:
  
   ```bash
   kubectl apply -f /path/to/your/deployment.yaml
   ```
3. Restart the services you want to remove instrumentation for.
```

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 6]
```diff
@@ -208,6 +270,26 @@ Run the following commands and restart the service to stop injecting the library
...
+
+**Note:** You can disable ASM code security while keeping APM up by adding the `DD_IAST_ENABLED=false` environment variable to your deployments.
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:289]:
```suggestion
<div class="alert alert-info"> You can disable ASM code security while keeping APM up by adding the <code>DD_IAST_ENABLED=false</code> environment variable to your deployments.</div>
```

File: content/en/security/application_security/enabling/single_step/code_security.md [hunk 7]
```diff
@@ -239,6 +321,17 @@ To stop producing traces, remove library injectors and restart the infrastructur
...
 
+{{% tab "Kubernetes" %}}
+
+1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
+2. Under `asm:`, remove `iast:` and all following configuration in`datadog-values.yaml`.
+3. Run the following command:
+   ```bash
+   helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
+   ```
+4. Restart your services.
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/code_security.md:332]:
```suggestion
3. Run the following command:
  
    ```bash
   helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
   ```
4. Restart your services.
```

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 8]
```diff
@@ -172,6 +173,68 @@ docker run -d --name dd-agent \
...
+   ```bash
+   helm install datadog-agent -f datadog-values.yaml datadog/datadog
+   ```
+5. Do a rolling restart on your applications for instrumentation to take effect.
+
+More information on Kubernetes Single Step Instrumentation here:
+
+* [Enabling or Disabling Single Step Instrumentation by namespaces][15]
+* [Specifying Instrumentation libraries versions][16]
+* [Removing Instrumentation on specific deployments][17]
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:223]:
```suggestion
1. Add the Helm Datadog repo:
  
    ```bash
    helm repo add datadog https://helm.datadoghq.com
    helm repo update
    ```
2. Create a Kubernetes Secret to store your Datadog [API key][10]:
  
   ```bash
   kubectl create secret generic datadog-secret --from-literal api-key=$DD_API_KEY
   ```
3. Create `datadog-values.yaml` and add the following configuration:
  
   ```
   datadog:
    apiKeyExistingSecret: datadog-secret
    site: <DATADOG_SITE>
    apm:
      instrumentation:
         enabled: true
    asm:
       threats:
         enabled: true
   ```
   Replace `<DATADOG_SITE>` with your [Datadog site][12].

4. Run the following command:
  
   ```bash
   helm install datadog-agent -f datadog-values.yaml datadog/datadog
   ```
5. Do a rolling restart on your applications for instrumentation to take effect.

For more information on Kubernetes single step Instrumentation, see the following:

* [Enabling or Disabling Single Step Instrumentation by namespaces][15]
* [Specifying Instrumentation libraries versions][16]
* [Removing Instrumentation on specific deployments][17]
```

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 9]
```diff
@@ -209,6 +272,25 @@ Run the following commands and restart the service to stop injecting the library
...
+         labels:
+           admission.datadoghq.com/enabled: "false"
+   ```
+2. Apply the configuration:
+   ```bash
+   kubectl apply -f /path/to/your/deployment.yaml
+   ```
+3. Restart the services you want to remove instrumentation for.
+
+**Note:** You can disable ASM while keeping APM up by adding the `DD_APPSEC_ENABLED=false` environment variable to your deployments.
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:291]:
```suggestion
1. Set the `admission.datadoghq.com/enabled:` label to `"false"` for the pod spec:
     
   ```yaml
   spec:
     template:
       metadata:
         labels:
           admission.datadoghq.com/enabled: "false"
   ```
2. Apply the configuration:
   
   ```bash
   kubectl apply -f /path/to/your/deployment.yaml
   ```
3. Restart the services you want to remove instrumentation for.

<div class="alert alert-info"> You can disable ASM while keeping APM up by adding the <code>DD_APPSEC_ENABLED=false</code> environment variable to your deployments.</div>
```

File: content/en/security/application_security/enabling/single_step/threat_detection.md [hunk 10]
```diff
@@ -241,6 +323,18 @@ To stop producing traces, remove library injectors and restart the infrastructur
...
+{{% tab "Kubernetes" %}}
+
+1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
+2. Under `asm:`, remove `threats:` and all following configuration in`datadog-values.yaml`.
+3. Run the following command:
+   ```bash
+   helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
+   ```
+4. Restart your services.
+
```
Comment (michaelcretzman) [inline content/en/security/application_security/enabling/single_step/threat_detection.md:335]:
```suggestion
1. Under `apm:`, remove `instrumentation:` and all following configuration in `datadog-values.yaml`.
2. Under `asm:`, remove `threats:` and all following configuration in`datadog-values.yaml`.
3. Run the following command:
   
   ```bash
   helm upgrade datadog-agent -f datadog-values.yaml datadog/datadog
   ```
4. Restart your services.

```

**Review by michaelcretzman — APPROVED** [review]
Approved with suggestions.

**Pull request merged**